### PR TITLE
Avoid going in critical when lid interface doesn't exist

### PR DESCRIPTION
**Pull request by augustin-b**
- /proc/acpi/button/lid/ could not exist on systems without LID switch (tested in VirtualBox VM)
- thus false when there is nothing to detect
- better indentation

### CODE REVIEW

File: src/Plug.vala [hunk 1]
```diff
@@ -378,20 +378,22 @@ namespace Power {
...
-            try {
-                var enumerator = interface_path.enumerate_children (
-                GLib.FileAttribute.STANDARD_NAME,
-                FileQueryInfoFlags.NONE);
-                FileInfo lid;
-                if ((lid = enumerator.next_file ()) != null) {
-                    debug ("Detected lid switch");
-                    return true;
-                }
+            if (interface_path.query_exists()) {
```
Comment (donadigo) [inline src/Plug.vala]: Space before brackets: `query_exists ()`.

Comment (augustin-b) [reply src/Plug.vala]: I wonder about that check in the backlight_detect.

The /sys/class/backlight is kernel sysfs interface which should be always present.
It only has entries when backlight driver is loaded.

File: src/Plug.vala [hunk 2]
```diff
@@ -378,20 +378,22 @@ namespace Power {
...
-            try {
-                var enumerator = interface_path.enumerate_children (
-                GLib.FileAttribute.STANDARD_NAME,
-                FileQueryInfoFlags.NONE);
-                FileInfo lid;
-                if ((lid = enumerator.next_file ()) != null) {
-                    debug ("Detected lid switch");
-                    return true;
-                }
+            if (interface_path.query_exists ()) {
```
Comment (kvdz) [inline src/Plug.vala]: use upower instead proc file system. /proc is deprecated and shouldn't be used anymore https://upower.freedesktop.org/docs/UPower.html
see properties

Comment (augustin-b) [reply src/Plug.vala]: OK, I'll try this https://upower.freedesktop.org/docs/UPower.html#UPower:LidIsPresent

**Comment by augustin-b** [comment]
@kvdz Thank's for advice.

I've just implemented it in #68 

Closing this...

**Pull request closed without merging**